src/rest/mod.rs

```diff
@@ -86,6 +86,13 @@ impl Rest {
                 path.push_str(&params);
             }
         }
+        #[cfg(feature = "optimized-access")]
+        let url = if R::OPTIMIZED_ACCESS_SUPPORTED {
```

Comment:
you could also do 

```
if cfg!(feature = "optimized-access") {
  // ...
} else {
  // ...
}
```

with this you could avoid intruducing the `OPTIMIZED_ACCESS_SUPPORTED` const.

Reply:
That looks nice! I wasn't aware of that syntax. 

I don't think you can avoid the OPTIMIZED_ACCESS_SUPPORTED constant though because only certain Rest endpoints are enabled for optimized access and the Request somehow has to communicate that its endpoint supports optimized access

Reply:
Is this roughly what you were thinking of?
```rust
        let url = if cfg!(feature = "optimized-access") {
            if R::OPTIMIZED_ACCESS_SUPPORTED {
                format!("{}{}", self.endpoint.optimized_access_rest(), path)
            } else {
                format!("{}{}", self.endpoint.rest(), path)
            }
        } else {
            format!("{}{}", self.endpoint.rest(), path)
        };
```
Unfortunately this doesn't compile because OPTIMIZED_ACCESS_SUPPORTED is only presented if the optimized-access feature is activated and the compiler doesn't seem to understand that it has to be present in this case.

Reply:
ah nevermind, I didn't understand why you need OPTIMIZED_ACCESS_SUPPORTED in the first place, but it's just because not all endpoints are supported yet.

Thanks for the PR